### PR TITLE
[consensus][qs] support OptQuorumStorePayload in consensus

### PR DESCRIPTION
**Pull request by ibalajiarun**
## Description

This PR adds support for processing proposals with OptQuorumStorePayload in consensus. Round manager ensures that the data corresponding to optimistic batches exist before signing the proposal. Otherwise, the proposal is enqueued until QS acknowledges data availability. 

Naively, this PR checks if optimistic batches are available locally and proceeds processing the proposal if so. If not, it wait until QS fetches both optimistic batches and **proof batches**.

### CODE REVIEW

File: consensus/src/block_storage/block_store.rs [hunk 1]
```diff
@@ -551,6 +551,15 @@ impl BlockStore {
             Duration::ZERO
         }
     }
+
+    pub async fn wait_for_payload(&self, block: &Block) -> anyhow::Result<()> {
+        self.payload_manager.get_transactions(block).await?;
```
Comment (zekun000) [inline consensus/src/block_storage/block_store.rs:556]: is there a timeout inside?

**Review by zekun000 — APPROVED** [review]

**Comment by sasha8** [comment]
> ## Description
> This PR adds support for processing proposals with OptQuorumStorePayload in consensus. Round manager ensures that the data corresponding to optimistic batches exist before signing the proposal. Otherwise, the proposal is enqueued until QS acknowledges data availability.
> 
> Naively, this PR checks if optimistic batches are available locally and proceeds processing the proposal if so. If not, it wait until QS fetches both optimistic batches and **proof batches**.

Why do we need to wait **proof batches**? Do you plan to change it in the next PRs?

**Comment by ibalajiarun** [comment]
> Why do we need to wait **proof batches**? Do you plan to change it in the next PRs?

This was easier to implement for now. We can discuss and change later. Basically, my assumption was fetching a proof batch is easier than fetching the optimistic batch.

**Review by sasha8 — APPROVED** [review]
LGTM

File: consensus/src/block_storage/block_store.rs (hunk 1 above)
Comment (ibalajiarun) [reply consensus/src/block_storage/block_store.rs]: added a timeout here for now. plan to refactor for heuristics later.

**Pull request merged**